go/config/config.go

```diff
@@ -340,6 +341,16 @@ func (config *ClientConfiguration) ToProtobuf() (*protobuf.ConnectionRequest, er
 		request.ConnectionTimeout = connectionTimeout
 	}
 
+	// Handle TLS root certificates
+	if config.AdvancedClientConfiguration.tlsConfig != nil {
```

Comment:
We should avoid using nested conditionals where possible. Consider using a helper function to consolidate these three tests, to simplify this logic test.